### PR TITLE
gin logging api

### PR DESCRIPTION
**Pull request by dragos-rebegea**

### CODE REVIEW

**Review by sstanculeanu — DISMISSED** [review]

File: aggregator/api/gin/webServer.go [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+	if ws.cancelFunc != nil {
+		ws.cancelFunc()
+	}
+
+	var err error
+	ws.Lock()
+	if ws.httpServer != nil {
+		err = ws.httpServer.Close()
+	}
+	ws.Unlock()
```
[the remark this answers is not in the document]
Comment (bogdan-rosianu) [reply aggregator/api/gin/webServer.go:113]: why RLock? it shuts down the server, it should definitely remain as it is

[the remark this answers is not in the document]
Comment (ssd04) [reply aggregator/api/gin/webServer.go:113]: ahh, right :+1:

**Review by sstanculeanu — APPROVED** [review]

File: aggregator/api/gin/webServer_test.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+		t.Parallel()
+
+		ws, err := NewWebServerHandler("127.0.0.1:8080")
+		assert.Nil(t, err)
+		assert.False(t, check.IfNil(ws))
+	})
+}
+
+func TestWebServer_StartHttpServer(t *testing.T) {
+	t.Run("upgrade on get returns error", func(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (dragos-rebegea) [reply aggregator/api/gin/webServer_test.go:25]: adding t.Parallel here will result in a data race

**Review by sstanculeanu — APPROVED** [review]

File: aggregator/api/gin/webServer_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (ssd04) [reply aggregator/api/gin/webServer_test.go:25]: i wasn't very explicit, i was referring to the main test only

**Review by ssd04 — APPROVED** [review]

**Pull request merged**